exporter/kafkaexporter/raw_marshaler.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafkaexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/kafkaexporter"
+
+import (
+	"errors"
+
+	"github.com/Shopify/sarama"
+	jsoniter "github.com/json-iterator/go"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+var errUnsupported = errors.New("unsupported serialization")
+
+type rawMarshaler struct {
+	marshaler jsoniter.API
+}
+
+func newRawMarshaler() rawMarshaler {
+	marshaler := jsoniter.Config{
+		SortMapKeys: true,
```

Comment:
This does impact performance, is sorted keys important here?

Reply:
Mostly to make my tests pass reliably. If we do move away from jsoniter, that might solve itself:
https://stackoverflow.com/questions/18668652/how-to-produce-json-with-sorted-keys-in-go
> The json package always orders keys when marshalling.